examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	handler "github.com/openfaas/templates-sdk/go-http"
+	"github.com/pelletier/go-toml"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const cfgPath = "/var/openfaas/secrets/vcconfig"
+
+var (
+	vsClt *vsClient // Persist vSphere connection.
+	once  sync.Once // For handleSignal() to be called once.
+	debug bool      // True for verbose logging.
+)
+
+// vcConfig represents the toml vcconfig file
+type vcConfig struct {
+	VCenter struct {
+		Server, User, Password string
+		Insecure               bool
+	}
+}
+
+// cloudEvent is a subsection of a Cloud Event.
+type cloudEvent struct {
+	Data types.Event `json:"data,omitempty"`
+}
+
+type vsClient struct {
+	govmomi *govmomi.Client
+	rest    *rest.Client
+	tagMgr  *tags.Manager
+}
+
+type vmConfig struct {
+	name    string
+	valBool bool
+	valInt  int
+}
+
+// Handle a function invocation
+func Handle(req handler.Request) (handler.Response, error) {
+	ctx := context.Background()
+	debug = setVerbosity()
+
+	// Load config every time, to ensure the most updated version is used.
+	cfg, err := loadConfig(cfgPath)
+	if err != nil {
+		return handlerResponseWithError("load vcconfig: %w",
+			http.StatusInternalServerError,
```

Comment:
nit: strictly speaking an invalid user provided config would not be 5xx. But keep it for simplicity.

Reply:
You're right. It's not a internal server error. The error can be caused by incorrect vcconfig.toml format (`400 Bad Request`) or wrong credentials (`401 Unauthorized`). 

I think I'll go with `401 Unauthorized` just because that's the whole point of those configs: so we can log into vSphere. 

I'll change it to `401 Unauthorized`. I wrote the error handling helper function so it's simple to pass in the status code.

Reply:
I'll also change it so that if they can't log into govmomi (maybe because of bad credentials), I'll return the `401 Unauthorized` status code for errors returned by the `newClient()` function.